commons/commons-serialization/src/main/kotlin/ac/uk/ebi/biostd/xml/XmlSerializer.kt

```diff
@@ -22,31 +24,46 @@ import com.fasterxml.jackson.dataformat.xml.XmlMapper
 import com.fasterxml.jackson.dataformat.xml.ser.ToXmlGenerator
 import ebi.ac.uk.model.Attribute
 import ebi.ac.uk.model.File
+import ebi.ac.uk.model.FilesTable
 import ebi.ac.uk.model.Link
+import ebi.ac.uk.model.LinksTable
 import ebi.ac.uk.model.Section
 import ebi.ac.uk.model.Submission
 import ebi.ac.uk.model.Table
+import ebi.ac.uk.util.collections.ifRight
 import org.xml.sax.InputSource
 import java.io.StringReader
 import javax.xml.parsers.DocumentBuilderFactory
 
 class XmlSerializer {
+    fun <T> serialize(element: T): String = mapper.writeValueAsString(element)
 
-    fun <T> serialize(element: T): String {
-        return mapper.writeValueAsString(element)
-    }
+    fun deserialize(value: String) = deserialize(value, Submission::class.java)
+
+    fun <T> deserialize(element: String, type: Class<out T>): T {
+        var deserialized: Any? = null
+        val xml = buildXmlFile(element)
+        val attributeDeserializer = AttributeXmlDeserializer(DetailsXmlDeserializer())
```

Comment:
I believe this is not necessary at all, (check json example) jackson object mapper is able to identify which deserializer call based on the class type